docs/StyleSpecificationGuide.md

```diff
@@ -0,0 +1,217 @@
+# Style Specification Guide
+
+![boston theme](./boston.png)
+
+harp.gl uses a custom specification to style the map's layers. harp.gl currently comes with three default theme variants. Click on a theme's name to take a look at it's style specification.
+* [Berlin base](../@here/harp-map-theme/resources/berlin_tilezen_base.json)
+* [Berlin reduced day](../@here/harp-map-theme/resources/berlin_tilezen_day_reduced.json)
+* [Berlin reduced night](../@here/harp-map-theme/resources/berlin_tilezen_night_reduced.json)
+
+## Top-level properties
+
+A harp.gl style specification has the following top-level properties:
+
+```json
+{
+   "sky": {},
+   "fog": {},
+   "clearColor": "#87959A",
+   "lights": [],
+   "defaultTextStyle": {},
+   "textStyles": [],
+   "fontCatalogs": [],
+   "images": {},
+   "poiTables": [],
+   "styles": {}
+}
+```
+
+### `styles`
+
+The `styles` object is where all the layer styling is defined (eg: water, highways, labels,earth, buildings).
+
+```json
+"styles": {
+   "tilezen": [
+      {
+         "description": "highway-label",
+         "when": "$layer == 'roads' && ((kind == 'highway') && ((kind_detail == 'motorway')))",
+         "technique": "text",
+         "attr": {
+            "color": "#000000",
+            "backgroundColor": "#FFFFFF",
+            "backgroundOpacity": 0.5,
+            "size": 16,
+            "priority": 25,
+            "fadeNear": 0.8,
+            "fadeFar": 0.9
+         },
+         "renderOrder": 10
+      }
+   ]
+}
+```
+
+Within `styles`, you must define a __style set name__. A style set name is a group of styles. This can be helpful if you want to apply a group of styles to an individual data source. For example, when adding a new datasource to the map, you can specify the style set name to be used:
+
+```javascript
+const omvDataSource = new harp.OmvDataSource({
+   baseUrl: "https://xyz.api.here.com/tiles/herebase.02",
+   apiFormat: harp.APIFormat.XYZOMV,
+   styleSetName: "tilezen", // <- style set name goes here
+   authenticationCode: 'YOUR-XYZ-TOKEN HERE', 
+});
+```
+
+You are free to have as many style set names as you would like.
+
+Within each style set name is where you can define style properties for layers. An individual style rule object has at least four required properties:
```

Comment:
The user of "layer" in this section is not really correct.
One style refers to a subset of  the data (defined by the condition). So a style could be a subset of a layer but could also work across layers (e.g. you could have one style for all points in the datasource)